sph/include/sph/particles_data_gpu.cuh

```diff
@@ -200,6 +206,12 @@ public:
         whd = DevVector<HydroType>(whdTable.begin(), whdTable.end());
     }
 
+    void uploadHelmEOS(const sph::helmholtz_constants::Helmholtz_EOS* helmEOS)
+    {
+        cudaMalloc(&devHelmEOS, sizeof(helmEOS));
+        cudaMemcpy(devHelmEOS, helmEOS, sizeof(helmEOS), cudaMemcpyHostToDevice);
```

Comment:
Memory leak. In practice only leaks at the end of the simulation, but will still show up in memory sanitize tools.
Better to allocate this in the translation unit that implements the Helmoltz EOS and make an object to control the lifetime, and put that into the propagator, like we do for e.g. gravity/multipole_holder.